crates/swc_css_prefixer/src/prefixer.rs

```diff
@@ -512,121 +460,123 @@ impl VisitMut for Prefixer {
         for mut n in take(&mut n.rules) {
             n.visit_mut_children_with(self);
 
-            new.append(&mut self.added_rules);
+            for mut n in take(&mut self.added_top_rules) {
+                let old_rule_prefix = self.rule_prefix.clone();
+
+                self.rule_prefix = Some(n.0);
+
+                n.1.visit_mut_children_with(self);
+
+                new.push(n.1);
+
+                self.rule_prefix = old_rule_prefix;
+            }
+
             new.push(n);
         }
 
+        // Avoid duplicate prefixed at-rules
+        new.dedup_by(|a, b| match (a, b) {
```

Comment:
This can be an issue if it's not sorted.
`dedup_by` only deduplicates if it's consecutive